datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -328,7 +328,9 @@ impl GroupedHashAggregateStream {
             .collect();
 
         let name = format!("GroupedHashAggregateStream[{partition}]");
-        let reservation = MemoryConsumer::new(name).register(context.memory_pool());
+        let reservation = MemoryConsumer::new(name)
```

Comment:
https://docs.rs/datafusion/latest/datafusion/execution/memory_pool/struct.MemoryConsumer.html#method.with_can_spill

I agree 

@kazuyukitanimura and @yjshen  do you concur? Is there any way we could potentially test this behavior?

Reply:
Can we create TestMemoryPool which will panic if unspillable operator is registered ?

Reply:
Yes, looks good.
Another idea to test is that to write a test aggregation job with small memory pool size that fails with the greedy pool but succeeds with the fair pool. Not sure if we can write such case...